basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package basiclibrary;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class LibraryTest {
+    @Test
+    public void testRoll() {
+        int[] arr = new int[4];
+        int length = arr.length;
+        assertEquals(
+                "This should check the length of the array.",
+                length,
+            Library.roll(4).length
+        );
+    }
+
+    @Test
+    public void testContainsDuplicates_false() {
+        int[] arr = new int[]{1,2,3};
+        assertEquals(
+                "This should return false as there are no duplicates in your array.",
```

Comment:
the way you word your assert message here and elsewhere is a perfect way to communicate what your test is doing